shared/selene/util/ssh/ssh.py

```diff
@@ -101,3 +105,32 @@ def connect(self):
     def disconnect(self):
         """Close ssh connection."""
         self.client.close()
+
+
+def validate_rsa_public_key(public_key: str) -> bool:
+    """Check the specified public key to determine if it is a well-formed RSA key.
+
+    According to the the specification, the first part of the key is a length-prefixed
+    string. The length is packed as a big-endian unsigned integer.  The expected value
+    is 7 because the following string, 'ssh-rsa', is 7 bytes long.
+
+    :param public_key: key to validate
+    :return: boolean indicating if validation check passed.
+    """
+    is_valid = False
+    public_key_parts = public_key.split()
+    if len(public_key_parts) == 3:
+        key_type, key, _ = public_key.split()
```

Comment:
I'm fairly sure the third component in an ssh-rsa key is optional. It's mostly used as an identifier of the creator, but officially is a `comment` so if:
`ssh-rsa ABC123 gez@local` was a valid key, then
`ssh-rsa ABC123` should also be valid

https://serverfault.com/questions/743548/what-significance-does-the-user-host-at-the-end-of-an-ssh-public-key-file-hold/743551

Reply:
nice catch.  fixed.